pyogrio/_io.pyx

```diff
@@ -31,20 +32,20 @@ log = logging.getLogger(__name__)
 # Mapping of OGR integer field types to Python field type names
 # (index in array is the integer field type)
 FIELD_TYPES = [
-    'int32',        # OFTInteger, Simple 32bit integer
-    None,           # OFTIntegerList, List of 32bit integers, not supported
-    'float64',      # OFTReal, Double Precision floating point
-    None,           # OFTRealList, List of doubles, not supported
-    'object',       # OFTString, String of UTF-8 chars
-    None,           # OFTStringList, Array of strings, not supported
-    None,           # OFTWideString, deprecated, not supported
-    None,           # OFTWideStringList, deprecated, not supported
-    'object',       #  OFTBinary, Raw Binary data
-    'datetime64[D]',# OFTDate, Date
-    None,           # OFTTime, Time, NOTE: not directly supported in numpy
-    'datetime64[s]',# OFTDateTime, Date and Time
-    'int64',        # OFTInteger64, Single 64bit integer
-    None            # OFTInteger64List, List of 64bit integers, not supported
+    'int32',         # OFTInteger, Simple 32bit integer
+    None,            # OFTIntegerList, List of 32bit integers, not supported
+    'float64',       # OFTReal, Double Precision floating point
+    None,            # OFTRealList, List of doubles, not supported
+    'object',        # OFTString, String of UTF-8 chars
+    None,            # OFTStringList, Array of strings, not supported
+    None,            # OFTWideString, deprecated, not supported
+    None,            # OFTWideStringList, deprecated, not supported
+    'object',        #  OFTBinary, Raw Binary data
+    'datetime64[D]', # OFTDate, Date
+    None,            # OFTTime, Time, NOTE: not directly supported in numpy
+    'datetime64[us]',# OFTDateTime, Date and Time
```

Comment:
Shouldn't this be `datetime64[ms]`?  I'm not familiar enough with the datetime types to know, but since we only have `ms` precision from GDAL, it seems like we would want to match that.